scripts/RelayHubGasEstimation.ts

```diff
@@ -54,374 +63,429 @@ function logTitle(title: string) {
 
-  printGasAnalysis(relayCallProcessFirstResult);
+  const tokenGas = !isSponsored
+    ? {
+        tokenGas: correctEstimatedCallCost(estimatedTokenPaymentGas.toNumber()),
+      }
+    : {};
 
-  console.log('ROUND 2');
-  let relayCallProcessSecondResult = await triggerRelayCallProcess({
-    message,
+  const completeReq: RelayRequest = cloneRelayRequest(relayRequest, {
+    request: {
+      data: encodedFunction,
+      to: token.address,
+      nonce: (await forwarder.nonce()).toString(),
+      tokenAmount: fees,
+      tokenContract,
+      gas: internalDestinationCallCost,
+      ...tokenGas,
+    },
   });
 
-  printGasAnalysis(relayCallProcessSecondResult);
+  return completeReq;
+}
 
-  logTitle('Gas prediction - without token payment:');
+async function estimateTokenTransferGasOverhead(fees: string) {
+  await deployAndSetup();
+  // refill SW balance
+  await token.mint('9000', forwarder.address);
+
+  const { forwarderInitialBalance, relayWorkerInitialBalance } =
+    await getCurrentBalances();
+
+  const transferReceiver = ethers.Wallet.createRandom();
+
+  // necessary to execute the transfer tx without relay
+  const balanceToTransfer = 1000;
+
+  // forge the request
+  const completeReq: RelayRequest = await forgeRequest(
+    transferReceiver.address,
+    balanceToTransfer,
+    fees,
+    token,
+    relayRequest,
+    forwarder
+  );
 
-  message = 'RIF ENVELOPING '.repeat(77);
-  relayCallProcessFirstResult = await triggerRelayCallProcess({
+  const { signature } = await getSuffixDataAndSignature(
+    forwarder,
+    completeReq,
+    owner
+  );
+
+  const relayCallResult = await relayHub
+    .connect(relayWorker)
+    .relayCall(completeReq, signature, {
+      gasLimit,
+      gasPrice,
+    });
+  const txReceipt = await relayCallResult.wait();
+
+  await assertRelayedTransaction(
+    forwarderInitialBalance,
+    relayWorkerInitialBalance,
+    BigNumber.from(balanceToTransfer),
+    BigNumber.from(fees)
+  );
+  await printGasStatus(txReceipt);
+}
+
+async function runGasPrediction(
+  messageSize: number,
+  runWithoutTokenPayment: boolean
+) {
+  const message = 'RIF ENVELOPING '.repeat(messageSize);
+
+  const relayCallProcessFirstResult = await triggerRelayCallProcess({
     message,
     estimateGasLimit: true,
-    noPayment: true,
-    noTokenGas: true,
+    noPayment: runWithoutTokenPayment,
+    noTokenGas: runWithoutTokenPayment,
   });
 
   printGasAnalysis(relayCallProcessFirstResult);
 
   console.log('ROUND 2');
-  relayCallProcessSecondResult = await triggerRelayCallProcess({
+
+  const relayCallProcessSecondResult = await triggerRelayCallProcess({
     message,
-    noPayment: true,
-    noTokenGas: true,
   });
 
   printGasAnalysis(relayCallProcessSecondResult);
-  
-  logTitle('Gas estimation tests for SmartWallet')
-  
-  message = 'RIF Enveloping '.repeat(32);
-  ({ recipient, forwarder, owner, relayRequest, relayHub } = await deployAndSetup());
+}
+
+async function runGasEstimation() {
```

Comment:
I'd remove this at all. It's not clear to me why we have it.

Reply:
Removed.